Hosting Dashboard Emails: Professional email

xavier-lc opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of DOTDASH-480

## Proposed Changes

* Add the professional email config step.
* I've tried to reuse most of the logic already in place on `/email/{siteSlug}`, but I had to adapt the code to the dashboard standards (queries, translate...).
* The main files to review that haven't been copied or are queries are:
  * client/dashboard/emails/add-professional-email/index.tsx
  * client/dashboard/emails/add-professional-email/components/mailbox-form.tsx
  * client/dashboard/emails/add-professional-email/components/mailbox-input.tsx

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* As part of or effort to modernize the hosting dashboard.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open the live preview
* Go to `/v2/emails/add-professional-email/{siteSlug}`
* Add multiple emails and proceed to checkout
* Submit the checkout
* Verify that you can access the titan mailbox for the newly created addresses

## Translation screenshots

- `Your password reset email is <strong>%(userEmail)s</strong>. <passwordChangeLink>Change it</passwordChangeLink>.`

<img width="1801" height="890" alt="Screenshot from 2025-10-21 15-35-29" src="https://github.com/user-attachments/assets/023b0521-e093-4182-8b42-248c7779602f" />

- `You can purchase new mailboxes at the prorated price of <strong>%(proratedPrice)s</strong> per mailbox.`
- `All of your mailboxes are due to renew at the regular price of <strong>%(fullPrice)s</strong> per mailbox when your subscription renews on <strong>%(expiryDate)s</strong>.`

<img width="1842" height="482" alt="image" src="https://github.com/user-attachments/assets/42eb0c4f-f9cf-4105-acb6-d67e83080910" />

- `You can purchase new mailboxes at the regular price of <strong>%(price)s</strong> per mailbox per month.`
- `You can purchase new mailboxes at the regular price of <strong>%(price)s</strong> per mailbox per year.`

<img width="1828" height="323" alt="image" src="https://github.com/user-attachments/assets/b7bf8cc1-3bd2-40e3-b1b4-575cb78f8cef" />

- `Try free today - renews at the regular price of %(cost)s per %(termLocalized)s (excl. taxes) when your free trial ends on %(endDate)s.`

<img width="1840" height="356" alt="image" src="https://github.com/user-attachments/assets/84516860-70e9-4e08-97a9-d974d11235b6" />

- `Try free today - first renewal at %(firstRenewalPrice)s (excl. taxes) after your free trial ends on %(endDate)s, regular price %(cost)s per %(termLocalized)s (excl. taxes).`

<img width="1840" height="356" alt="image" src="https://github.com/user-attachments/assets/ecf07342-183f-493f-8f0a-d6e802adbd74" />

- `Add as many mailboxes as you need. Each one has a price of %(cost)s per %(termLocalized)s (excl. taxes).`

<img width="1840" height="356" alt="image" src="https://github.com/user-attachments/assets/f7dbde1c-540a-417a-8bd1-8c59ccb06872" />

- `You cannot add emails to this domain: %(errors)s.`
- `You cannot add emails to this domain.`

<img width="838" height="230" alt="image" src="https://github.com/user-attachments/assets/c42e9989-7dc7-459d-b47c-34d464df1fff" />

- `This email address must have a different domain than <strong>%(domain)s</strong>. Please use a different email address.`

<img width="1879" height="462" alt="image" src="https://github.com/user-attachments/assets/698d3322-b6b1-4a80-9304-d9702cc1a81e" />

- `Please use unique email addresses. <strong>%(emailAddress)s</strong> already exists in your account.`

<img width="1879" height="462" alt="image" src="https://github.com/user-attachments/assets/54e6a069-f25e-4811-8c3c-360568b72d30" />

- `<strong>%(mailbox)s</strong> is not available: %(message)s`

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Add products query

Commit:
Add route

Commit:
Add card & buttons

Commit:
Add `MailboxForm`

Commit:
Mute suffix

Commit:
Disable on submit

Commit:
Add `IntervalLength` type

Commit:
Get the interval from the URL search

Commit:
Add `getProductSlugForProviderAndInterval`

Taken from `client/my-sites/email/email-providers-comparison/stacked/provider-cards/selectors/get-product-by-provider-and-interval.ts`

Commit:
Determine product on submit

Commit:
Import `MailboxForm` entity from `client/my-sites/email/form/mailboxes/index.ts`

Commit:
Use `EmailProvider` from api-core

Commit:
Add `sanitizeMailboxValue` util

Commit:
Add export

Commit:
Fix types

Commit:
Add `createNewMailbox`

Commit:
Wiring the form and the entity

Commit:
Add `MailboxOperations`

Copied from `client/my-sites/email/form/mailboxes/components/utilities/mailbox-operations.ts`

Commit:
Extracted `MailboxInput` with errors

Commit:
Add `current_user_cannot_add_email_reason` to `Domain`

Commit:
Bind password reset as well

Commit:
Fix password reset visibility

Commit:
Prevent submission if not valid

Commit:
Show `current_user_cannot_add_email_reason` errors

Commit:
Add `getEmailProductProperties`

Copied from `client/my-sites/email/form/mailboxes/components/utilities/get-email-product-properties.ts`

Commit:
Add `Product`type

Commit:
Add `default` case to make ts happy

Commit:
Add constants

Commit:
Added `getCartItems`

Copied from `client/my-sites/email/form/mailboxes/components/utilities/get-cart-items.ts`

Commit:
Handle shopping cart submit

Commit:
Prevent error

Commit:
Added `EmailNonDomainOwnerNotice`

Copied from `client/my-sites/email/email-non-domain-owner-message/index.tsx`

Commit:
Fix

Commit:
Fix URLs

Commit:
Add mailbox working

Commit:
Add remove form

Commit:
Dispatch field state so validation happens while typing

Commit:
Fix lint error

Commit:
Add `form` element

Commit:
Add offer text

Commit:
Merge branch 'trunk' into DOTDASH-480-1

xavier-lc commented on client/dashboard/domains/hooks/use-domain-owner-username.ts:
```diff
@@ -0,0 +1,39 @@
+	login: string;
+}
+
+export function useDomainOwnerUserName(
```
Function taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/components/data/query-domain-owner-username/index.ts#L19

xavier-lc commented on client/dashboard/emails/add-professional-email/components/email-non-domain-owner-notice.tsx:
```diff
@@ -0,0 +1,106 @@
+	return params.toString();
+};
+
+export const EmailNonDomainOwnerNotice = ( props: EmailNonDomainOwnerMessageProps ) => {
```
Code taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/my-sites/email/email-non-domain-owner-message/index.tsx#L27

mmtr replied:
For **non-super admins**, this does not work when trying to access a domain of a site you don't have access because the API returns a token error:

<img width="1440" height="449" alt="Screenshot 2025-10-17 at 13 44 36" src="https://github.com/user-attachments/assets/b6591c5a-ddff-4a86-9620-2947d037bdcf" />

xavier-lc commented on client/dashboard/emails/constants/index.ts:
```diff
@@ -0,0 +1,2 @@
+export const TITAN_MAIL_MONTHLY_SLUG = 'wp_titan_mail_monthly';
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/packages/calypso-products/src/constants/titan.ts#L1

mmtr replied:
Can you use `TitanMailSlugs` from `@automattic/api-core` instead?

See https://github.com/Automattic/wp-calypso/blob/a743e6b1d9f148e907937fabef3dcf63fe4c2526/packages/api-core/src/constants.ts#L209-L212

xavier-lc commented on client/dashboard/emails/entities/constants.ts:
```diff
@@ -0,0 +1,22 @@
+export const FIELD_PASSWORD_RESET_EMAIL = 'passwordResetEmail';
+export const FIELD_UUID = 'uuid';
+
+export type TitanMailboxFields = {
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/my-sites/email/form/mailboxes/constants.ts#L15

xavier-lc commented on client/dashboard/emails/entities/mailbox-form.ts:
```diff
@@ -0,0 +1,285 @@
+} from './types';
+import type { Validator } from './validators';
+
+class MailboxForm< T extends SupportedEmailProvider > {
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/my-sites/email/form/mailboxes/index.ts#L37

xavier-lc commented on client/dashboard/emails/entities/mailbox-operations.ts:
```diff
@@ -0,0 +1,64 @@
+import { MailboxForm } from './mailbox-form';
+import { SupportedEmailProvider } from './types';
+
+export class MailboxOperations {
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/my-sites/email/form/mailboxes/components/utilities/mailbox-operations.ts#L6

xavier-lc commented on client/dashboard/emails/entities/types.ts:
```diff
@@ -0,0 +1,172 @@
+
+type FieldError = TranslateResult | null;
+
+interface MailboxFormField< Type > {
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/my-sites/email/form/mailboxes/types.ts#L21

xavier-lc commented on client/dashboard/emails/entities/validators.ts:
```diff
@@ -0,0 +1,410 @@
+import wp from 'calypso/lib/wp';
+import type { FieldError, MailboxFormFieldBase } from './types';
+
+interface Validator< T > {
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/my-sites/email/form/mailboxes/validators.ts#L8

xavier-lc commented on client/dashboard/emails/utils/get-cart-items.ts:
```diff
@@ -0,0 +1,103 @@
+
+const getGSuiteCartItems = () => {};
+
+export const getCartItems = (
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/my-sites/email/form/mailboxes/components/utilities/get-cart-items.ts#L14

xavier-lc commented on client/dashboard/emails/utils/get-email-product-properties.ts:
```diff
@@ -0,0 +1,46 @@
+	quantity: number;
+};
+
+const getEmailProductProperties = (
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/my-sites/email/form/mailboxes/components/utilities/get-email-product-properties.ts#L15

xavier-lc commented on client/dashboard/emails/utils/get-product-slug-for-provider-and-interval.ts:
```diff
@@ -0,0 +1,24 @@
+/**
+ * Returns the correct product slug for the specified provider and interval using a map.
+ */
+export const getProductSlugForProviderAndInterval = (
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/my-sites/email/email-providers-comparison/stacked/provider-cards/selectors/get-product-by-provider-and-interval.ts#L17

xavier-lc commented on client/dashboard/emails/utils/is-titan-monthly-product.ts:
```diff
@@ -0,0 +1,6 @@
+import { Product } from '@automattic/api-core';
+import { TITAN_MAIL_MONTHLY_SLUG } from '../constants';
+
+export function isTitanMonthlyProduct( titanMailProduct: Product ): boolean {
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/lib/titan/is-titan-monthly-product.ts#L4

xavier-lc commented on client/dashboard/emails/utils/sanitize-mailbox-value.ts:
```diff
@@ -0,0 +1,4 @@
+export const sanitizeMailboxValue = ( suggestedValue: string ): string =>
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/my-sites/email/form/mailboxes/components/utilities/sanitize-mailbox-value.ts#L1

xavier-lc commented on client/dashboard/shopping-cart/errors.ts:
```diff
@@ -0,0 +1,12 @@
+export class CartActionError extends Error {
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/packages/shopping-cart/src/errors.ts#L1

xavier-lc commented on client/dashboard/shopping-cart/types.ts:
```diff
@@ -0,0 +1,1229 @@
+import type { CartActionError } from './errors';
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/packages/shopping-cart/src/types.ts#L1

xavier-lc commented on client/dashboard/utils/domain.ts:
```diff
@@ -143,6 +145,19 @@ export function hasTitanMailWithUs( domain: SiteDomain | FullDomain ) {
+ * subscription must have at least one mailbox, `1` is the default return value even for domains
+ * without an active Titan subscription.
+ */
+export function getMaxTitanMailboxCount( domain: SiteDomain | FullDomain ): number {
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/lib/titan/get-max-titan-mailbox-count.ts#L8

xavier-lc commented on client/dashboard/utils/domain.ts:
```diff
@@ -143,6 +145,19 @@ export function hasTitanMailWithUs( domain: SiteDomain | FullDomain ) {
+	return ( domain.titan_mail_subscription as TitanEmailSubscription )?.maximum_mailbox_count ?? 1;
+}
+
+export function getGSuiteMailboxCount( domain: SiteDomain | FullDomain ): number {
```
Taken from https://github.com/Automattic/wp-calypso/blob/127f23fb25d8dcad4aa2c8ecba220c7522bff6ff/client/lib/gsuite/get-gsuite-mailbox-count.js#L1

Commit:
Add missing getEmailManagementPath

Commit:
Reuse `buildQueryString`

xavier-lc commented on client/dashboard/utils/email-paths.ts:
```diff
@@ -257,3 +258,27 @@ export const getEmailCheckoutPath = (
 	siteName ? `/mailboxes/${ siteName }` : '/mailboxes';
+
+// Generates URL: /email/:domain/manage/:site
+export const getEmailManagementPath: EmailPathUtilityFunction = (
```
Taken from https://github.com/Automattic/wp-calypso/blob/2da474a7d17149d603a38ebdd8a8e0194ac3bd41/client/my-sites/email/paths.ts#L175

mmtr replied:
Emails shouldn't be managed anymore in `/email/:domain/manage/:site` but in `/v2/emails?domainName=:domain`

xavier-lc replied:
Why not `/v2/emails/:domain`? Do we have an issue for that?

mmtr replied:
Because the `?domainName=` param is part of the DataViews filter. See https://github.com/Automattic/wp-calypso/pull/106413

rcrdortiz replied:
We could support both implementations if we think it's valuable.

xavier-lc replied:
Addressed on https://github.com/Automattic/wp-calypso/pull/106524/commits/0ae1e83befb199e3426aadea2e30a33a6020282d

Commit:
Navigate out on domain not found

Commit:
Preload queries

Commit:
Merge branch 'trunk' into DOTDASH-480-1

mmtr commented on client/dashboard/emails/add-professional-email/index.tsx:
```diff
@@ -0,0 +1,303 @@
+					? sprintf(
+							// Translators: %(cost)s is the displayed cost, %(termLocalized)s is the localized term (e.g. "year"), %(endDate)s is the date the trial ends (e.g. "October 26, 2005").
+							__(
+								'Add as many mailboxes as you need. Each one will renew at the regular price of %(cost)s per %(termLocalized)s (excl. taxes) when your free trial ends on %(endDate)s.'
```
This does not consider the discounted price of the first renewal:

Current Calypso (prod) | Hosting Dashboard (this PR)
--- | ---
<img width="1075" height="883" alt="Screenshot 2025-10-17 at 13 17 40" src="https://github.com/user-attachments/assets/21846a1f-2c60-4df8-9f7f-4ef1fceeb7d5" /> | <img width="713" height="517" alt="Screenshot 2025-10-17 at 13 17 31" src="https://github.com/user-attachments/assets/e308936e-0402-41de-8d36-76bf65efc750" />

xavier-lc replied:
Addressed on https://github.com/Automattic/wp-calypso/pull/106524/commits/439cc089cbd4b2b4753458f93c5253edb63ca7c7

mmtr commented on client/dashboard/emails/add-professional-email/index.tsx:
```diff
@@ -0,0 +1,303 @@
+	const hasOffer =
+		product?.introductory_offer &&
+		product?.introductory_offer.interval_count > 0 &&
+		product?.introductory_offer?.interval_unit;
```
The offer is only valid if it has not been used before, so you need to check `domain.titan_mail_subscription.is_eligible_for_introductory_offer` too.

See https://github.com/Automattic/wp-calypso/blob/279b271e235d1c634addd187c04d6dd702d6907b/client/lib/titan/is-domain-eligible-for-titan-free-trial.ts#L6

xavier-lc replied:
Good catch! Addressed on https://github.com/Automattic/wp-calypso/pull/106524/commits/2fff14be55e936f5dda6a23306d208bfe58c5e07.

mmtr commented on client/dashboard/emails/add-professional-email/index.tsx:
```diff
@@ -0,0 +1,303 @@
+							),
+							{
+								cost: product.combined_cost_display,
+								termLocalized: product.product_term_localized,
```
This does not consider the prorated cost for users who are not using the free trial. See https://github.com/Automattic/wp-calypso/blob/279b271e235d1c634addd187c04d6dd702d6907b/client/my-sites/email/email-pricing-notice/index.tsx#L57-L71

<img width="967" height="655" alt="Screenshot 2025-10-17 at 13 28 01" src="https://github.com/user-attachments/assets/e57ffdad-1f90-4d1b-8b93-3a7355d8cba9" />

xavier-lc replied:
Thanks, addressed on https://github.com/Automattic/wp-calypso/pull/106524/commits/ded392a4d639ef3a8f57148c92ad4bf428b82dd1

<img width="3681" height="2168" alt="image" src="https://github.com/user-attachments/assets/3fe58524-398a-4c7c-94c0-8508f9bd9a29" />

mmtr replied:
> All of your mailboxes are due to renew at the regular price of €28.53 per mailbox

That seems wrong? I thought the regular price was €38, so maybe €28.53 is a prorated price?

mmtr replied:
FWIW, I don't know how to get that message. This is what I see which it's indeed the correct message:

<img width="706" height="586" alt="Screenshot 2025-10-20 at 12 31 10" src="https://github.com/user-attachments/assets/b0588e0a-0f31-4a92-992c-16f4860eb9a2" />

xavier-lc replied:
I'm seeing the same price in production, have you checked what you see for `testmmtr25business.blog`? Maybe it's a difference between a `.blog` or `.com` domain 🤔

<img width="3840" height="1773" alt="image" src="https://github.com/user-attachments/assets/2c033785-0ff4-4129-800d-a745f5c93500" />

mmtr replied:
Yeah, I see the same message in prod and this branch:

<img width="967" height="655" alt="Screenshot 2025-10-17 at 13 28 01" src="https://github.com/user-attachments/assets/e57ffdad-1f90-4d1b-8b93-3a7355d8cba9" />


Anyway, since you can see the "regular price of €28.53" part in prod too, we can tackle that in a follow-up if needed if we/someone determines that it's actually a bug.

mmtr commented on client/dashboard/emails/add-professional-email/index.tsx:
```diff
@@ -0,0 +1,303 @@
+	const { data: products } = useQuery( productsQuery() );
+	const productSlug = getProductSlugForProviderAndInterval( 'titan', interval );
+	const product = products?.[ productSlug ] as Product;
+	const { data: site } = useQuery( siteBySlugQuery( domainName ) );
```
One site can have more than one domain, so you actually need to fetch the site attached to the domain:

```
const { data: site } = useQuery( siteByIdQuery( domain.blog_id ) );
```

mmtr replied:
Since https://github.com/Automattic/wp-calypso/pull/106555 also needs a site, I fixed this and created a shared util there. See https://github.com/Automattic/wp-calypso/pull/106555/commits/1c0e7b58bdedea3efdcc492419e3f2b0f3089571

xavier-lc replied:
Addressed on https://github.com/Automattic/wp-calypso/pull/106524/commits/527e2586d3200d84da88ef9742a1a8448f37c8de

mmtr commented on client/dashboard/emails/utils/get-product-slug-for-provider-and-interval.ts:
```diff
@@ -0,0 +1,24 @@
+		case 'google_workspace':
+			return intervalLength === 'monthly'
+				? 'wp_google_workspace_business_starter_monthly'
+				: 'wp_google_workspace_business_starter_yearly';
```
Can you use `GoogleWorkspaceSlugs` from `@automattic/api-core` instead?

See https://github.com/Automattic/wp-calypso/blob/a743e6b1d9f148e907937fabef3dcf63fe4c2526/packages/api-core/src/constants.ts#L214-L216

xavier-lc replied:
Totally missed that they were already there, addressed on https://github.com/Automattic/wp-calypso/pull/106524/commits/6ac2db36a23d3cfbf2227ec75e6c64728538d33b

rcrdortiz commented on client/dashboard/app/router/emails.tsx:
```diff
@@ -104,6 +107,35 @@ export const chooseEmailSolutionRoute = createRoute( {
+
+		const domain = await queryClient.ensureQueryData( domainQuery( domainName ) );
+		const mailboxAccounts = await queryClient.ensureQueryData(
+			mailboxAccountsQuery( domain?.blog_id, domainName )
```
`blog_id` isn't nullable, we can just use `domain.blog_id` instead.

xavier-lc replied:
Addressed on https://github.com/Automattic/wp-calypso/pull/106524/commits/785c667fc2d5f57bbf95569a1a2034d9457f7eae

mmtr commented on client/dashboard/emails/add-professional-email/components/email-non-domain-owner-notice.tsx:
```diff
@@ -0,0 +1,96 @@
+export const EmailNonDomainOwnerNotice = ( props: EmailNonDomainOwnerMessageProps ) => {
+	const { domain, selectedSite, source } = props;
+
+	const ownerUserName = useDomainOwnerUserName( selectedSite, domain );
```
Can we simplify this with `domain.owner`?

<img width="529" height="281" alt="Screenshot 2025-10-17 at 14 05 55" src="https://github.com/user-attachments/assets/d2a19c82-5d62-49c0-8371-a4be8a93f807" />

xavier-lc replied:
The original function has a comment regarding Jetpack shenanigans: 

> Due to Jetpack sites overriding the user.ID with a completely different thing, when Jetpack overrides this property, the original WordPress.com user Id ends stored as user.linked_user_ID, so in those cases, that's the ID we have to use.

Do you think it's safe to rely on `owner` in that case as well?

mmtr replied:
Yes, it should be safe. The issue is only present when fetching users from `/sites/:site/users`, since it may return a list of users with an incorrect user ID on Jetpack sites, so we need to match `domainSubscription.userId` with `user.linked_user_ID`.

The `/domain-details/:domain` endpoint don't have that problem, because it does not fetch users from `/sites/:site/users`, it uses the WP.com DB users table (see fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Snqzva%2Qcyhtvaf%2Sqbznvaf%2Sqbznva%2Qznantrzrag%2Qqbznva.cuc%3Se%3Q996212oq%23505-og)

xavier-lc replied:
Addressed on https://github.com/Automattic/wp-calypso/pull/106524/commits/7cbbcc132b26478f17c35378f235e267931003da

rcrdortiz commented on client/dashboard/emails/add-professional-email/index.tsx:
```diff
@@ -0,0 +1,303 @@
+					  ) }
+			</Text>
+
+			<form onSubmit={ handleSubmit }>
```
Should we prevent default behaviour here?
````
e.preventDefault();
````

xavier-lc replied:
Sure, added on https://github.com/Automattic/wp-calypso/pull/106524/commits/4cddc24280bd64a7bd5da16816efbaad0e960d30

mmtr commented on client/dashboard/emails/add-professional-email/components/email-non-domain-owner-notice.tsx:
```diff
@@ -0,0 +1,96 @@
+	const buildLoginUrl = () => {
+		const redirectUrlParameter =
+			source === 'email-comparison'
+				? getPurchaseNewEmailAccountPath( selectedSite?.slug, domain?.domain, '', 'login-redirect' )
```
Maybe unrelated to this PR, but in the new HD, emails for a given domain are purchased from `/v2/emails/choose-email-solution/:domain`

Commit:
Merge branch 'trunk' into DOTDASH-480-1

Commit:
Consider `is_eligible_for_introductory_offer` for titan offer

mmtr commented on client/dashboard/emails/add-professional-email/index.tsx:
```diff
@@ -0,0 +1,301 @@
+	const disabled = isSubmitting || showEmailPurchaseDisabledMessage;
+
+	let endDate = new Date();
+	const hasOffer = domain && isDomainEligibleForTitanIntroductoryOffer( { domain, product } );
```
`isDomainEligibleForTitanIntroductoryOffer` already checks whether domain is defined, so I think you can simplify this:

 ```suggestion
	const hasOffer = isDomainEligibleForTitanIntroductoryOffer( { domain, product } );
```

mmtr replied:
Actually, you need to change the signature of that function from `domain: Domain;` to `domain: Domain | undefined`;

mmtr replied:
Since the domain is now fetched via a suspense query in https://github.com/Automattic/wp-calypso/pull/106555, I think you can leave this as-is, and I'll remove the domain check in that PR (see b3723a8fb3bedaf511ad88730e87cf67554736fa)

xavier-lc replied:
I added it because of type complaints, it's very annoying that the query result can be `undefined` at all times 🫠

Commit:
Added `PricingNotice`

Copied from `client/my-sites/email/add-mailboxes/email-provider-pricing-notice.tsx`

Commit:
Use `siteByIdQuery` instead of `siteBySlugQuery`

Commit:
Use constants from api-core

Commit:
Remove unneeded nullish coalescing

Commit:
preventDefault on submit

xavier-lc commented:
> [can be a follow-up]
> When I have an email account in my cart, I can't add more mailboxes. I'm getting a notice about it, but I have no way of navigating to the cart from the current screen. Maybe we could make `shopping cart` a link to the shopping cart so that it's easier for our users to access it.

Good catch. I'd rather have it as a follow-up so this doesn't get out of hand anymore.

> I also think that the main bar should be showing a cart too right?

Yup, that's what calypso currently does. I don't know if we can currently do that on the dashboard, will have to check.

mmtr commented:
In the Figma design, there is a footer with the overall cost of all mailboxes:

<img width="969" height="609" alt="Screenshot 2025-10-20 at 12 16 29" src="https://github.com/user-attachments/assets/0d23d540-cf29-41ee-a85d-852815db26bd" />

Are we implementing that in a follow-up?

mmtr commented on client/dashboard/emails/add-professional-email/index.tsx:
```diff
@@ -0,0 +1,259 @@
+			numberOfMailboxes
+		);
+
+		const checkoutPath = getEmailCheckoutPath(
+			domainName,
+			domain.domain,
+			router.state.location.pathname,
+			mailboxOperations.mailboxes[ 0 ].getAsCartItem().email
+		);
```
This why @rcrdortiz  sees the site selector [here](https://github.com/Automattic/wp-calypso/pull/106524#issuecomment-3415416650).

It can be reproduced on secondary domains:

- Purchase two domains for a given site
- Pick one of them and make it the primary domain
- If you go to `/v2/emails/add-professional-email/:primaryDomain`, you can proceed to checkout normally
- If you go to `/v2/emails/add-professional-email/:secondaryDomain`, the checkout fails and you are redirected to the site selector

I think it can be fixed by using the site slug in the checkout URL:

```suggestion
		const checkoutPath = getEmailCheckoutPath(
			site.slug,
			domain.domain,
			router.state.location.pathname,
			mailboxOperations.mailboxes[ 0 ].getAsCartItem().email
		);
```

xavier-lc replied:
@mmtr @rcrdortiz you'r right! Fixed on https://github.com/Automattic/wp-calypso/pull/106524/commits/5b9eb7b6975278c510879c1896d540e0d00d2848

Commit:
Update `getEmailManagementPath`

Commit:
Drop `useDomainOwnerUserName` in favor of `domain?.owner`

Commit:
Add first year discount text

Commit:
Redirect to emails dashboard if domain details can't be loaded

xavier-lc commented:
> In the Figma design, there is a footer with the overall cost of all mailboxes. Are we implementing that in a follow-up?

Yup, I'd do it after this one is merged so it doesn't get more complicated.

Commit:
Fix `getEmailCheckoutPath` use

Commit:
Fix commented code

Commit:
Add shopping cart link to the error notice when needed

mmtr reviewed (approved):
I think we can merge this for now so we keep things moving, and address the following remaining feedback in follow-ups:
- Unauthorized access to domains (https://github.com/Automattic/wp-calypso/pull/106524#discussion_r2439598324 and https://github.com/Automattic/wp-calypso/pull/106524#issuecomment-3422426374)
- Implement footer with price summary and continue button (https://github.com/Automattic/wp-calypso/pull/106524#issuecomment-3415379073 and https://github.com/Automattic/wp-calypso/pull/106524#issuecomment-3421433277)

Commit:
Fix lint error

Commit:
Merge branch 'trunk' into DOTDASH-480-1

Commit:
Fix css

Commit:
Fix spacing

Commit:
Fix lint

mmtr commented:
> For some reason I'm now redirected to the login page rather than to `/v2/emails` 🤔
> 
>  Screen.Recording.2025-10-20.at.16.50.10.mov

Looks like this is caused by https://github.com/Automattic/wp-calypso/pull/106494
